python/cuda_cccl/cuda/cccl/parallel/experimental/algorithms/_radix_sort.py

```diff
@@ -242,3 +242,43 @@ def radix_sort(
         A callable object that can be used to perform the merge sort
     """
     return _RadixSort(d_in_keys, d_out_keys, d_in_values, d_out_values, order)
+
+
+def radix_sort(
+    d_in_keys: DeviceArrayLike | DoubleBuffer,
+    d_out_keys: DeviceArrayLike | None,
+    d_in_values: DeviceArrayLike | DoubleBuffer | None,
+    d_out_values: DeviceArrayLike | None,
+    order: SortOrder,
+    num_items: int,
+    begin_bit: int | None = None,
+    end_bit: int | None = None,
+    stream=None,
+):
+    import numpy as np
+    from numba.cuda import device_array
+
+    sorter = make_radix_sort(d_in_keys, d_out_keys, d_in_values, d_out_values, order)
+    tmp_storage_bytes = sorter(
+        None,
+        d_in_keys,
+        d_out_keys,
+        d_in_values,
+        d_out_values,
+        num_items,
+        begin_bit,
+        end_bit,
+        stream,
+    )
+    tmp_storage = device_array(shape=(tmp_storage_bytes,), dtype=np.uint8)
```

Comment:
important: I'm not sure about this. Subsequent call is asynchronous. This means that we can exist this function before the algorithm even starts running on GPU. I don't think we hold on to the `tmp_storage` reference to guarantee this is sound. When we discussed this interface, I suggested `cuda{Malloc,Free}Async`. Is there any way we can rely on `cuda.bindings.runtime.cudaMallocAsync` here instead of `numba`? Same comment applies to other algorithms.

Reply:
OK, I introduced a lightweight container that wraps `cudaMallocAsync/Free`. It gives a nice speedup over using numba, but it is still a tiny bit slower than using cupy arrays.

https://github.com/NVIDIA/cccl/pull/5207/commits/fd3eab4f74cb86a4d664e98b720739cdafd0203b

I would request that @leofang bless this mini-container implementation before merging. 

As a side note, I tried using `cuda.core.experimental.Buffer` rather than `cuda.bindings.runtime.cudaMallocAsync` directly, but it was much slower.

Reply:
I strongly recommend that we use `cuda.core` and not reinvent the wheel. This is not the first time we are rushing into a workaround within the team instead of getting things fixed properly. The path to fix is pretty clear: https://github.com/NVIDIA/cuda-python/issues/756. Let's discuss this in the team sync-up later today.

Reply:
I am resolving this conversation. We got to the bottom of the issue making `Buffer` allocations slow (it's that the default pool has a release threshold of 0).

I have updated the top-level PR description to describe our current memory allocation strategy.